### ISSUE TITLE
Add a way to ignore directory or files

### ISSUE THREAD

**Issue by JJ** [markdownlint/markdownlint#368]
## Motivation

As an author, I want to be able to exclude files or directories in a per-file or repo-wide basis, so that I can focus my attention on errors reported on files that I am actually using.

There does not seem to be a way to exclude files or directories at the rule or at the file level. The use case is one in which this linter is added to a repository, and there are some files that are no longer used, but have not been deleted either. An "attic" kind of directory where those could be placed so that they are not linted would be a great addition.

## Specification

- A command that ignores files in `.mdl.rb`, or
- YAML front matter for files that need to be ignored, or
- a .lintignore file, following the implementation of .gitignore, for instance

## Downstream Impact

Not sure, since there's no change over existing implementations.

**Comment by rfay** [comment]
Duplicate of #167 I think

**Comment by JJ** [comment]
Definitely related, except this one is a bit wider and the proposals are a bit more concrete.

**Comment by jamesquilty** [comment]
Definitely a duplicate of #167.

> * A command that ignores files in .mdl.rb, or

 Probably `.mldrc` was meant?

> * YAML front matter for files that need to be ignored  

 CommonMark-compliant parsers will parse YAML front matter, unfortunately, otherwise this might be a good solution.

**Comment by btonasse** [comment]
Yeah, unfortunately markdownlint is not usable for me without this feature.
